### PR TITLE
Support value types/ienumerable in MemberDataAttributeBase

### LINKED ISSUE #1288: Support value types in `MemberDataAttributeBase`
### Problem

[`MemberDataAttributeBase`](https://github.com/xunit/xunit/blob/bccfcccf26b2c63c90573fe1a17e6572882ef39c/src/xunit.core/MemberDataAttributeBase.cs) does not support collection of value types in the result of the member.
A member returning `IEnumerable<int>` will throw an `ArgumentException` in [`GetData`](https://github.com/xunit/xunit/blob/bccfcccf26b2c63c90573fe1a17e6572882ef39c/src/xunit.core/MemberDataAttributeBase.cs#L71) because it is impossible to cast the `IEnumerable<int>` into `IEnumerable<object>`.

This issue may be a problem when creating new implementation of `MemberDataAttributeBase`.

### Steps to reproduce

```csharp
public sealed class SingleMemberDataAttribute : ImprovedMemberDataAttributeBase
{
    public SingleMemberDataAttribute(string memberName, params object[] parameters)
        : base(memberName, parameters) { }

    protected override object[] ConvertDataItem(MethodInfo testMethod, object item)
    {
        return new object[] { item };
    }
}

private static IEnumerable<int> DataProviderMethod()
{
    yield return 1;
    yield return 2;
}

[Theory]
[SingleMemberData(nameof(DataProviderMethod))]
public void ValueTypeTestWithAttribute(int value)
{
    // ...
}
```

### Suggested solution

In [`GetData`](https://github.com/xunit/xunit/blob/bccfcccf26b2c63c90573fe1a17e6572882ef39c/src/xunit.core/MemberDataAttributeBase.cs#L69-L73), replace the lines:
```csharp
var dataItems = obj as IEnumerable<object>;
if (dataItems == null)
    throw new ArgumentException($"Property {MemberName} on {type.FullName} did not return IEnumerable<object>");

return dataItems.Select(item => ConvertDataItem(testMethod, item));
```

by:
```csharp
var dataItems = obj as IEnumerable;
if (dataItems == null)
    throw new ArgumentException($"Property {MemberName} on {type.FullName} did not return IEnumerable");

return dataItems.Cast<object>().Select(item => ConvertDataItem(testMethod, item));
```

### PR DESCRIPTION
**Pull request by hughbe**
Fixes #1288

### CODE REVIEW

File: src/xunit.core/MemberDataAttributeBase.cs [hunk 1]
```diff
@@ -66,11 +67,11 @@ public override IEnumerable<object[]> GetData(MethodInfo testMethod)
             if (obj == null)
                 return null;
 
-            var dataItems = obj as IEnumerable<object>;
+            var dataItems = obj as IEnumerable;
             if (dataItems == null)
                 throw new ArgumentException($"Property {MemberName} on {type.FullName} did not return IEnumerable<object>");
```
Comment (onovotny) [inline src/xunit.core/MemberDataAttributeBase.cs]: Nit: Exception message stills says `IEnumerable<object>`, should now just be `IEnumerable`

**Comment by hughbe** [comment]
@bradwilson does this look good to you?

**Comment by hughbe** [comment]
Friendly ping @bradwilson :)

**Comment by bradwilson** [comment]
Danke!

**Pull request merged**